### PR TITLE
[jit] Fix partially shared constrained calls in llvmonly mode.

### PR DESCRIPTION
**Pull request by vargaz**
Fixes https://github.com/mono/mono/issues/7535.

### CODE REVIEW

**Comment by marek-safar** [comment]
@kumpera could you please review

File: mono/mini/method-to-ir.c [hunk 1]
```diff
@@ -8519,7 +8519,11 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 						MONO_EMIT_NEW_BRANCH_BLOCK (cfg, OP_IBEQ, is_ref_bb);
 
 						/* Non-ref case */
-						nonbox_call = (MonoInst*)mini_emit_calli (cfg, fsig, sp, addr, NULL, NULL);
+						if (cfg->llvm_only)
```
Comment (kumpera) [inline mono/mini/method-to-ir.c]: Shouldn't we just change `mini_emit_calli` to call `emit_llvmonly_calli` when `cfg->llvm_only` is true? There are a few other callsites in method-to-ir.c that don't use `emit_llvmonly_calli'.

**Comment by luhenry** [comment]
@monojenkins backport to 2018-02

**Comment by luhenry** [comment]
@monojenkins backport to 2017-12

**Pull request merged**